pandas/core/groupby/generic.py

```diff
@@ -1335,7 +1343,7 @@ def aggregate(self, func=None, *args, engine=None, engine_kwargs=None, **kwargs)
                 else:
                     # GH#32040, GH#35246
                     # e.g. test_groupby_as_index_select_column_sum_empty_df
-                    result = cast(DataFrame, result)
+                    result = cast(self._obj_1d_constructor, result)
```

Comment:
_constructor, not _1d_constructor.  Also i think this is OK as is

Reply:
I have also wrestled with whether results should stick to the subclass or just be DataFrames

Reply:
What does mypy resolve `self._constructor` to? I'd be surprised if it wasn't `Any`. You can check by adding `reveal_type(self._obj_1d_constructor)` on it's own line and then run mypy.